### PR TITLE
Add new consent values

### PR DESCRIPTION
**Pull request by arielf-camacho**
### Description

* Includes new controls for Unity v2018, v2019 and v2020 to handle GDPR consent.
* Includes new controls for Unity v2018, v2020 to handle new COPPA values (as it was only made on v2019).
* To accomplish the above, also includes the native API calls to make the buttons change the GDPR values in Android SDK, and retro-compatible changes for iOS.

### Link to Documentation Changes

None.

### Link to Issue or Ticket

[Trello Card](https://trello.com/c/T6zwoHd9/819-change-unity-binding-to-match-new-gdpr-consent)

### Checklist:

- [X] My code follows the style guidelines of this project. See [CONTRIBUTING.md](https://github.com/scalemonk/mediation-sdk-android/blob/master/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] New and existing UI tests pass locally with my changes. See [Mediation SDK Tests](https://github.com/scalemonk/mediation-sdk-tests)

### CODE REVIEW

File: 2017-Container/Assets/Demo/ScaleMonkDebugMenu.cs [hunk 1]
```diff
@@ -19,10 +19,16 @@ public class ScaleMonkDebugMenu : MonoBehaviour
         public Button ShowRewardedVideoButton;
         public Button ShowBannerButton;
         public Button StopBannerButton;
+        public Button CoppaForChild;
```
Comment (rodrilemos) [inline 2017-Container/Assets/Demo/ScaleMonkDebugMenu.cs:22]: Add `Button` at the end of those names

File: 2019-Container/Assets/Demo/ScaleMonkDebugMenu.cs [hunk 2]
```diff
@@ -21,11 +21,14 @@ public class ScaleMonkDebugMenu : MonoBehaviour
         public Button StopBannerButton;
         public Button CoppaForChild;
         public Button CoppaForNonChild;
+        public Button GrantGdprConsent;
```
Comment (rodrilemos) [inline 2019-Container/Assets/Demo/ScaleMonkDebugMenu.cs:24]: Use `Button` at the end of the name

File: 2019-Container/ProjectSettings/AndroidResolverDependencies.xml [hunk 3]
```diff
@@ -29,7 +29,7 @@
     <setting name="packageDir" value="Assets/Plugins/Android" />
     <setting name="patchAndroidManifest" value="True" />
     <setting name="patchMainTemplateGradle" value="True" />
-    <setting name="projectExportEnabled" value="True" />
+    <setting name="projectExportEnabled" value="False" />
```
Comment (rodrilemos) [inline 2019-Container/ProjectSettings/AndroidResolverDependencies.xml:32]: You can rollback this

File: 2020-Container/Assets/Demo/ScaleMonkDebugMenu.cs [hunk 4]
```diff
@@ -19,6 +19,11 @@ public class ScaleMonkDebugMenu : MonoBehaviour
         public Button ShowRewardedVideoButton;
         public Button ShowBannerButton;
         public Button StopBannerButton;
+        public Button CoppaForChild;
```
Comment (rodrilemos) [inline 2020-Container/Assets/Demo/ScaleMonkDebugMenu.cs:22]: Add `Button` at the end of those names

File: Source/Source/AdsEditorBinding.cs [hunk 5]
```diff
@@ -132,11 +133,16 @@ public bool AreInterstitialsEnabled()
...
         }
-
+        
+        [Obsolete("Use \"void SetHasGDPRConsent(GdprConsent status)\" method instead.")]
         public void SetHasGDPRConsent(bool consent)
         {
         }
 
+        public void SetHasGDPRConsent(GdprConsent consent)
+        {
```
Comment (rodrilemos) [inline Source/Source/AdsEditorBinding.cs:143]: We may need to log something here, so at least we have some feedback on the editor

**Review by rodrilemos — APPROVED** [review]

**Review by christiangelone — COMMENTED** [review]
Shouldnt we touch the .m file too 🤔 ?

File: 2019-Container/ProjectSettings/AndroidResolverDependencies.xml (hunk 3 above)
Comment (arielf-camacho) [reply 2019-Container/ProjectSettings/AndroidResolverDependencies.xml:32]: Will do, I don't know why Unity does this auto-generated.

File: 2017-Container/Assets/Demo/ScaleMonkDebugMenu.cs (hunk 1 above)
Comment (arielf-camacho) [reply 2017-Container/Assets/Demo/ScaleMonkDebugMenu.cs:22]: Done!

File: 2019-Container/Assets/Demo/ScaleMonkDebugMenu.cs (hunk 2 above)
Comment (arielf-camacho) [reply 2019-Container/Assets/Demo/ScaleMonkDebugMenu.cs]: Done!

File: 2019-Container/ProjectSettings/AndroidResolverDependencies.xml (hunk 3 above)
Comment (arielf-camacho) [reply 2019-Container/ProjectSettings/AndroidResolverDependencies.xml]: Done!

File: 2020-Container/Assets/Demo/ScaleMonkDebugMenu.cs (hunk 4 above)
Comment (arielf-camacho) [reply 2020-Container/Assets/Demo/ScaleMonkDebugMenu.cs:22]: Done!

File: Source/Source/AdsEditorBinding.cs (hunk 5 above)
Comment (arielf-camacho) [reply Source/Source/AdsEditorBinding.cs:143]: Done!

**Comment by arielf-camacho** [comment]
@christiangelone I touched the .m making calls to the native version that is old, because on iOS we have not yet implemented the new GDPR consent.

File: Source/Source/iOS/AdsiOSBinding.cs [hunk 6]
```diff
@@ -62,11 +64,20 @@ public bool AreInterstitialsEnabled()
...
         {
-            SMSetHasGDPRConsent(consent);
+            // Using this implementation until iOS native one is in place
+            SetHasGDPRConsent(consent);
+        }
+
+        public void SetHasGDPRConsent(GdprConsent consent)
+        {
+            // Using the above implementation until iOS native one is in place
+            SMSetHasGDPRConsent(consent ? GdprConsent.Granted : GdprConsent.NotGranted);
```
Comment (christiangelone) [inline Source/Source/iOS/AdsiOSBinding.cs:77]: be careful if enum are considered ints and start from zero then this code is broken 
   ` 0 ?  0 : 1`

**Review by christiangelone — CHANGES_REQUESTED** [review]

File: Source/Source/iOS/AdsiOSBinding.cs [hunk 7]
```diff
@@ -185,15 +196,20 @@ public void RewardedNotReady()
...
 
         [DllImport("__Internal")]
         private static extern void SMSetApplicationChildDirectedStatus(int status);
         
+        [Obsolete("Use \"SMSetHasGDPRConsent(int consent)\" method instead.")]
         [DllImport("__Internal")]
         private static extern void SMSetHasGDPRConsent(bool consent);
         
+        [DllImport("__Internal")]
+        private static extern void SMSetHasGDPRConsent(int consent);
```
Comment (christiangelone) [inline Source/Source/iOS/AdsiOSBinding.cs:211]: i dont see this method implemented

**Review by christiangelone — CHANGES_REQUESTED** [review]

File: Source/Source/iOS/AdsiOSBinding.cs (hunk 6 above)
Comment (arielf-camacho) [reply Source/Source/iOS/AdsiOSBinding.cs:77]: OK, will include the cast. Thanks!

File: Source/Source/iOS/AdsiOSBinding.cs (hunk 7 above)
Comment (arielf-camacho) [reply Source/Source/iOS/AdsiOSBinding.cs]: Removed!

**Review by christiangelone — APPROVED** [review]

**Review by rodrilemos — APPROVED** [review]

**Pull request merged**